### PR TITLE
Support for voice messages

### PR DESCRIPTION
**Pull request by Puncher1**
## Summary

This PR adds support for voice messages. 
Related PR: https://github.com/discord/discord-api-docs/pull/6082

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/permissions.py [hunk 1]
```diff
@@ -262,7 +262,7 @@ def text(cls) -> Self:
            Added :attr:`create_public_threads`, :attr:`create_private_threads`, :attr:`manage_threads`,
            :attr:`send_messages_in_threads` and :attr:`use_external_stickers` permissions.
         """
```
Comment (diceroll123) [inline discord/permissions.py:264]: might wanna add versionchanged

Comment (Puncher1) [reply discord/permissions.py:267]: Thanks

File: discord/flags.py [hunk 2]
```diff
@@ -450,6 +450,14 @@ def silent(self):
         """
         return 4096
 
+    @alias_flag_value
```
[the remark this answers is not in the document]
Comment (Puncher1) [reply discord/flags.py]: Ah yeah, was a left-over from copy-pasting the above one 😄, thanks!

File: discord/message.py [hunk 3]
```diff
@@ -183,6 +183,14 @@ class Attachment(Hashable):
         Whether the attachment is ephemeral.
 
         .. versionadded:: 2.0
+    duration: Optional[:class:`float`]
+        The duration of the audio file. Returns ``None`` if it's not a voice message.
```
Comment (Rapptz) [inline discord/message.py:187]:
```suggestion
        The duration of the audio file in seconds. Returns ``None`` if it's not a voice message.
```

File: discord/utils.py [hunk 4]
```diff
@@ -628,6 +628,12 @@ def _bytes_to_base64_data(data: bytes) -> str:
     return fmt.format(mime=mime, data=b64)
 
 
+def _base64_to_bytes(data: str) -> bytes:
+    encoded = data.encode('ascii')
+    bytes_ = b64decode(encoded)
+    return bytes_
```
Comment (Rapptz) [inline discord/utils.py:634]:
```suggestion
    return b64decode(data.encode('ascii'))
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
Minor nits.

**Comment by Puncher1** [comment]
It's officially documented now.

**Review by Rapptz — APPROVED** [review]

**Pull request merged**